src/main/java/matching_system/MatcherUITemplateJames.java

```diff
@@ -0,0 +1,86 @@
+package matching_system;
+
+
+import email_request.MatcherUIController;
+import email_request.MatcherUIPresenter;
+import email_request.MatcherUIResponseModel;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MatcherUITemplateJames extends JPanel implements ActionListener {
+    MatcherUIController matcherUIController;
+    private String requesterEmail;
+    private Map<Integer, String> targetUserEmails = new HashMap<>();
+
+    public MatcherUITemplateJames(MatcherUIController matcherUIController,
+                             String requesterEmail,
+                             MatchResponseModel matchResponseModel){
+
+        this.matcherUIController = matcherUIController;
+        this.requesterEmail = requesterEmail;
+        Integer count = 0;
+        for(UserData user: matchResponseModel.getMatchedData()){
+
+            String userIntro = "<html>Name: " + user.name + "<br>Gender: " +
+                    user.gender.getGender() + " age: " + user.age + " height: "
+                    + user.height + "<br> Hobby: " + user.hobby + " Program: " + user.programOfStudy +
+                    "<br>SelfIntro: " + user.selfIntro;
+            JLabel text = new JLabel(userIntro);
+            text.setAlignmentX(Component.CENTER_ALIGNMENT);
+            this.targetUserEmails.put(count, user.email);
+            JButton requestEmail = new JButton("Request Email");
+            requestEmail.setActionCommand("action" + count);
+            JPanel buttons = new JPanel();
+            buttons.add(requestEmail);
+            requestEmail.addActionListener(this);
+            this.add(text);
+            this.add(buttons);
+            this.setBorder(BorderFactory.createLineBorder(Color.black));
+            this.setPreferredSize(new Dimension(300, 125));
+            this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
```

Comment:
This line can be removed since we are setting layout on line 49 anyways.

In order to restore the division line, we can either add a lable like "--------------------" at the end of the loop or figure somthing out using Graphic.drawLine. Adding a lable is probably the most simple way.

Reply:
added the division lines between different profiles